socorro/unittest/external/fs/test_tar_crashstore.py

```diff
@@ -1,85 +1,143 @@
-        )
-        mocked_file_handle.close.assert_called_once_with()
-        tar_store.tar_file.add.assert_called_once_with(
-            '/tmp/091204bd-87c0-42ba-8f58-554492141212.jsonz',
-            '09/12/091204bd-87c0-42ba-8f58-554492141212.jsonz'
+        # this try to get it back
+        tar_store.close()
+
+        result_tar_fp = tarfile.TarFile('/tmp/a_tarball_name.tar')
+        result_gzip_fp = gzip.GzipFile(
+            fileobj=result_tar_fp.extractfile(
+                '091204bd-87c0-42ba-8f58-554492141212.jsonz'
+            )
         )
-        tar_store.os_module.unlink.assert_called_once_with(
-            '/tmp/091204bd-87c0-42ba-8f58-554492141212.jsonz'
+        reconstituted_processed_crash_as_str = result_gzip_fp.read().strip()
+
+        self.assertEqual(
+            processed_crash_as_str,
+            reconstituted_processed_crash_as_str
         )
 
 
+#==============================================================================
+class TestTarFileSequentialReadingCrashStorage(TestCase):
+
+    #--------------------------------------------------------------------------
+    def setUp(self):
+        super(TestTarFileSequentialReadingCrashStorage, self).setUp()
+
+    #--------------------------------------------------------------------------
+    def _get_config(self):
+        config = DotDict()
+        config.logger = Mock()
+        config.tarball_name = '/tmp/a_tarball_name.tar'
+        config.tarfile_module = tarfile
+        config.gzip_module = gzip
+        config.redactor_class = Redactor
+        config.forbidden_keys = ''
+
+        return config
+
+    #--------------------------------------------------------------------------
+    def test_get_processed(self):
+        config = self._get_config()
+        config.tarfile_module = tarfile
+        config.gzip_module = gzip
+
+        writing_tar_store = TarFileWritingCrashStore(config)
+
+        processed_crash_1 = {
+            'crash_id': '091204bd-87c0-42ba-8f58-554492141212',
+            'payload': 'nothing to see here',
+            'some_date': "1960-05-04T15:10:00"
+        }
+        processed_crash_2 = {
+            'crash_id': '666884bd-87c0-42ba-8f58-554492141212',
+            'payload': 'nothing to see here, ether',
+            'some_date': "1974-05-04T15:10:00"
+        }
+
+        writing_tar_store.save_processed(processed_crash_1)
+        writing_tar_store.save_processed(processed_crash_2)
+        writing_tar_store.close()
+
+        reading_tar_store = TarFileSequentialReadingCrashStore(config)
+
+        reconstituted_processed_crash_1 = reading_tar_store.get_processed(
+            'it does not matter what is said here'
+        )
+        eq_(processed_crash_1, reconstituted_processed_crash_1)
+
+        reconstituted_processed_crash_2 = reading_tar_store.get_processed(
+            'it does not matter what is said here'
+        )
+        eq_(processed_crash_2, reconstituted_processed_crash_2)
```

Comment:
Why are you doing two crashes? What does that test that 1 crash doesn't? If there's an explanation, perhaps a comment would be good.